server/src/main/java/io/druid/metadata/SQLMetadataRuleManager.java

```diff
@@ -377,7 +376,7 @@ public Void inTransaction(Handle handle, TransactionStatus transactionStatus) th
         return false;
       }
     }
-
+    poll();
```

Comment:
May throw exception if DB connection is disrupted. The repeating runnable in start() handles such a case.